### PR TITLE
store parsed versions and location in ES

### PR DESCRIPTION
**Pull request by snopoke**
@czue

### CODE REVIEW

File: testapps/test_pillowtop/tests/test_xform_pillow.py [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
+
 from corehq.apps.change_feed import topics
 from corehq.apps.change_feed.consumer.feed import change_meta_from_kafka_message
 from corehq.apps.change_feed.tests.utils import get_test_kafka_consumer
 from corehq.apps.es import FormES
 from corehq.form_processor.interfaces.processor import FormProcessorInterface
 from corehq.form_processor.utils import TestFormMetadata
 from corehq.form_processor.tests.utils import FormProcessorTestUtils
-from corehq.pillows.xform import XFormPillow, get_sql_xform_to_elasticsearch_pillow
+from corehq.pillows.xform import XFormPillow, get_sql_xform_to_elasticsearch_pillow, transform_xform_for_elasticsearch
```
Comment (dimagimon) [inline testapps/test_pillowtop/tests/test_xform_pillow.py]: E501 line too long (118 > 115 characters)

File: testapps/test_pillowtop/tests/test_xform_pillow.py [hunk 2]
```diff
@@ -70,3 +73,67 @@ def test_xform_pillow_sql(self):
...
         self.assertEqual('XFormInstance', form_doc['doc_type'])
+
+
+class TransformXformForESTest(SimpleTestCase):
+    def test_transform_xform_for_elasticsearch_app_versions(self):
+        doc_dict = {
+            'domain': 'demo',
+            'form': {
+                'meta': {
+                    'appVersion': 'CommCare Android, version "2.27.2"(414569). App v56. CommCare Version 2.27. Build 414569, built on: 2016-04-28'
```
Comment (dimagimon) [inline testapps/test_pillowtop/tests/test_xform_pillow.py]: E501 line too long (146 > 115 characters)

**Comment by czue** [comment]
:+1: on :white_check_mark:

**Comment by czue** [comment]
@dannyroberts @gcapalbo this is the change for the xforms index

File: corehq/ex-submodules/couchforms/datatypes.py [hunk 3]
```diff
@@ -1,4 +1,10 @@
 from collections import namedtuple
 
 
-GeoPoint = namedtuple('GeoPoint', 'latitude longitude altitude accuracy')
+class GeoPoint(namedtuple('GeoPoint', 'latitude longitude altitude accuracy')):
+    @property
+    def lat_long(self):
```
Comment (dannyroberts) [inline corehq/ex-submodules/couchforms/datatypes.py]: nit: `long` here versus `lon` below could be confusing

Comment (snopoke) [inline corehq/ex-submodules/couchforms/datatypes.py]: typo, thanks updated

File: corehq/pillows/mappings/case_mapping.py [hunk 4]
```diff
@@ -2,7 +2,11 @@
 from corehq.pillows.mappings import NULL_VALUE
 from corehq.util.elastic import es_index
 
+##################
+# NOTE to the next person who updates this name:
+#    You should also remove the CasePillow.get_unique_id method
```
Comment (esoergel) [inline corehq/pillows/mappings/case_mapping.py]: Haha, nice.  I was gonna say that `TODO` will be missed.  This won't, though.

**Comment by snopoke** [comment]
:white_check_mark:

**Pull request merged**